### DISCUSSION TITLE
Data navigation and menu restructuring

### DISCUSSION THREAD

**Discussion by ghost** [apache/superset#20878] [Ideas]
Hi folks, 

Want to start a discussion around making some improvements to the way we present data entities in our navigation and general information architecture.

### User problem
Through user research and product feedback, we've found that users do not understand the difference between a database and a dataset or the role of the dataset in visualization creation.

This is in part due to the current information architecture, which presents databases as the primary data entity in the product. Although connecting a database is a key first step in a user’s journey, it is an infrequent action. Instead, the dataset should be the central data object in the product as it is the basis for all visualizations.

<img width="274" alt="Screen Shot 2022-07-26 at 3 42 29 PM" src="https://user-images.githubusercontent.com/90715511/181124711-2ca2f7b1-8ac0-47bb-9861-520f1fd9932d.png">

<img width="752" alt="Screen Shot 2022-07-26 at 3 42 40 PM" src="https://user-images.githubusercontent.com/90715511/181124731-a5951669-aa6a-4dc1-bb8c-bad91730a840.png">

### Proposed solution
(All mock ups are for illustrative purposes only)

<img width="809" alt="Screen Shot 2022-07-26 at 3 51 37 PM" src="https://user-images.githubusercontent.com/90715511/181126439-8b812249-c8f6-43a2-ab47-641dec4c1491.png">

1. Move **Databases** to its own standalone CRUD view accessible under the manage section of the “Settings” menu
![Frame 378](https://user-images.githubusercontent.com/90715511/181126501-707939f5-756b-436c-b269-6ab06da45ec5.png)

2. Move **Datasets** to its own standalone CRUD view accessible from the main navigation
![Frame 379](https://user-images.githubusercontent.com/90715511/181126516-72ed3ceb-f8a3-4f0f-8448-00a285ce792d.png)

3. Rename remaining tabbed “Data” CRUD view (which would only have Saved Queries & Query History) to “**SQL**” (also rename main nav dropdown label)
![Frame 380](https://user-images.githubusercontent.com/90715511/181126548-d07f7cca-75a0-4a01-a76a-1c3deeeea867.png)

4. Add **Datasets** to the + Create menu in the top nav

When user has no databases connected:
<img width="428" alt="Screen Shot 2022-07-26 at 4 06 26 PM" src="https://user-images.githubusercontent.com/90715511/181127492-7fa23952-619a-40ad-a6bd-48fbbe5a9ff6.png">

When user has at least one database connected:
<img width="446" alt="Screen Shot 2022-07-26 at 4 06 43 PM" src="https://user-images.githubusercontent.com/90715511/181127509-8e008eca-9e66-45a1-8e67-027804a26772.png">

**Comment by mistercrunch** [comment]
This looks great to me. Much more comprehensive than the current navigation.

Suggesting one extra improvement -> on the dataset creation form, we could add a setting "cog" icon or similar right of the database connection dropdown list with a tooltip "manage database connections" that would link to the database connection page.
<img width="625" alt="Screen Shot 2022-08-01 at 5 22 34 PM" src="https://user-images.githubusercontent.com/487433/182266336-303eaecc-c8cb-49a0-8883-d2087ebabe26.png">

**Comment by ktmud** [comment]
These changes all make sense to me. One small proposal: now that "Datasets" has it's own top-level page and the top tabs are free for graph, how about we use it as a quick filter for Physical vs Virtual datasets?

<img width="879" alt="image" src="https://user-images.githubusercontent.com/335541/182297614-3a27ad51-83b5-4d09-872b-7186666fbbf9.png">

**Reply by ghost** [reply]
We've actually been discussing moving away from explicit Physical vs. Virtual labeling of datasets as it's confusing to users, and instead being more explicit about the "source" of a dataset (database, schema, table or SQL query).

**Reply by ktmud** [reply]
Could you clarify how would each source you mentioned would be configured and used? I'm not sure what a "schema only" or "database only" dataset would look like.

**Reply by ghost** [reply]
I think generally we want to do a better job of providing information about a dataset when the user is trying to locate or use it (so primarily in the dataset CRUD view and when selecting a dataset to use in Explore). We've found that telling a user that a dataset is "physical" or "virtual" at those points isn't so useful to them, and instead what they want to see is "this dataset is made of data from x table in x database" or "here's the SQL used to create this dataset". So just getting a little more specific about the source of a dataset rather than using Superset-specific terminology.

**Comment by cccs-rc** [comment]
Happy to see the "Databases" button moving to the Settings menu! For us, the dev team manage database connections and users ideally never even have to look at them, so this works well.

**Comment by codyml** [comment]
This seems like a great improvement!  One thought: could we go even farther with **4. Add Datasets to the + Create menu in the top nav**?  We could make "Dataset" into a top-level Create option, mirroring the structure of main navigation, and hide everything database-related behind the "Data" submenu (or even remove it completely to emphasize that database management shouldn't be a common operation).  And, if the user hasn't connected a database, it wouldn't make sense for them to be creating dashboards/charts/datasets at all, right?  So maybe the whole menu could disappear in that case, or it could contain exclusively links to database-connecting options.

**Reply by ghost** [reply]
We are going to be doing some smart hiding and showing in the Create menu – if users don't have a db connected, they will see "Connect database" in the + menu, but if they do have a db connected they will see "Add dataset" in the + menu. 

I think we want to leave the chart/dashboard options in that menu even when the user doesn't have their own data connected, though, because new users often use our example content to evaluate the tool.
